### PR TITLE
Dev 3753

### PR DESCRIPTION
**Pull request by msantos-sw**
Implemented methods that will be used by backend project.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -110,10 +133,34 @@ Money.prototype.toString = function () {
...
 };
 
-Money.prototype.format = function () {};
-
 Money.sum = function (...m) {
     return m.reduce((a, b) => a.plus(b));
 };
 
+Money.max = function(...m) {
+    if (!m[0] instanceof Money) throw new Error('invalid argument');
```
Comment (gilad-deel) [inline index.js:141]: you need to surround this with brackets

File: index.js [hunk 2]
```diff
@@ -110,10 +133,34 @@ Money.prototype.toString = function () {
...
+    
+    const currency = m[0].currency;
+
+    return m.reduce((acc, a) => {
```
Comment (gilad-deel) [inline index.js:145]: no need to pass `null` nor check it after. Please read about `reduce` syntax

File: index.js [hunk 3]
```diff
@@ -110,10 +133,34 @@ Money.prototype.toString = function () {
...
+        if (!(a instanceof Money) || currency !== a.currency) throw new Error('invalid argument');
+
+        if (acc === null || Money(a).gt(acc)) { acc = a; }
```
Comment (gilad-deel) [inline index.js:148]: https://github.com/letsdeel/backend/blob/dev/coding.md#blocks

**Review by gilad-deel — CHANGES_REQUESTED** [review]

File: index.js [hunk 4]
```diff
@@ -23,6 +23,11 @@ function Money(amount, currency) {
     }
 }
 
+Money.USD = function (v) {
+    if (v === null || v === undefined || v instanceof Money) throw new Error('invalid argument');
```
[the remark this answers is not in the document]
Comment (msantos-sw) [reply index.js]: fixed

File: index.js [hunk 5]
```diff
@@ -79,21 +80,43 @@ Money.prototype.minus = Money.prototype.sub = function (m) {
     return new Money(this.amount.minus(m.amount), this.currency);
 };
 
-Money.prototype.mul = Money.prototype.times = function (v) {
+Money.prototype.mul = Money.prototype.times = Money.prototype.multipliedBy = function (v) {
+    if (v === null || v === undefined || v instanceof Money) throw new Error('invalid argument');
```
[the remark this answers is not in the document]
Comment (msantos-sw) [reply index.js]: fixed

File: index.js [hunk 6]
```diff
@@ -79,21 +80,43 @@ Money.prototype.minus = Money.prototype.sub = function (m) {
...
+Money.prototype.abs = function() { 
+    return new Money(this.amount.abs(), this.currency);
+};
+
+Money.prototype.toNumber = function() { 
+    return this.amount.toNumber();
+};
+
+Money.prototype.isZero = function() {
+    return this.amount.toNumber() === 0;
```
[the remark this answers is not in the document]
Comment (msantos-sw) [reply index.js]: fixed

File: index.js [hunk 7]
```diff
@@ -79,21 +80,43 @@ Money.prototype.minus = Money.prototype.sub = function (m) {
...
+};
+
+Money.prototype.isPositive = function() { 
+    return this.amount.toNumber() >= 0;
```
[the remark this answers is not in the document]
Comment (msantos-sw) [reply index.js]: fixed

File: index.js (hunk 1 above)
Comment (msantos-sw) [reply index.js]: fixed

File: index.js (hunk 2 above)
Comment (msantos-sw) [reply index.js:161]: fixed

File: index.js (hunk 3 above)
Comment (msantos-sw) [reply index.js]: fixed

**Pull request merged**